core/modules/layout/layout.module

```diff
@@ -459,7 +476,7 @@ function layout_path_load($arg_value, $menu_item_name, $path_parts, $arg_positio
   $contexts = $menu_item->getContexts();
   $context_info = layout_get_context_info($contexts[$arg_position]->plugin);
   if (isset($context_info['load callback']) && function_exists($context_info['load callback'])) {
-    return $context_info['load callback']($arg_value);
+    return $context_info['load callback'](array('id' => $arg_value));
```

Comment:
I don't see the reason for passing this in as an array instead of just the ID like it was previously. This necessitated the creation of `layout_context_node_load()` and other wrappers, which seems unnecessary if we're only ever passing in an `'id'` parameter.

Reply:
> if we're only ever passing in an 'id' parameter.

@quicksketch that is actually the heart of the matter. In [layout::getContexts()](https://github.com/backdrop/backdrop/pull/3120/files#diff-f9a46040a1020182a9f049bb8c6864e8R778) we would want to call the load function for a context. This implementation allows us to pass in array of settings from the context form. Otherwise both core and contrib context forms would be limited to storing only a single identifier (the ID), limiting us to only simple context concepts.

Reply:
@quicksketch any thoughts?

Reply:
> Otherwise both core and contrib context forms would be limited to storing only a single identifier (the ID), limiting us to only simple context concepts.

Makes sense.

Reply:
While I see the need to handle more complex loading arguments, the addition of `layout_context_node_load()`, `layout_context_user_load()`, etc make it clear that this results in an API change.

I can think of two work-arounds to this problem.

We could special-case saving the context settings that if there is only a single setting and it's name is "id", then we save `$context->settings` as that single value instead of as an array. Then calling `$context_info['load callback']($context->settings)` would result in passing a single parameter that is compatible with `node_load()`, `user_load()`, etc.

Or instead of invoking the callback directly we could call it with `call_user_func_array()`, which would convert each context setting into an individual argument:
```
call_user_func_array($context_info['load callback'], $context->settings);
```

This would work in a way that was backwards compatible, but it would result in the settings array key order having meaning. If your load callback was `entity_load()` for example, a settings array of `array('entity_type' => 'node', 'id' => 1)` would call `entity_load('node', 1)` :+1:, but if the array were saved as `array('id' => 1, 'entity_type' => 'node')`, it would call `entity_load(1, 'node')` :-1:, resulting in an exception.

This might be fine though. IMO it's better than forcing all new load functions everywhere that must take only a single parameter that is always an array. If a developer wanted to take a single array that was keyed, they could structure their settings in a nested array in their settings form.